src/clr-angular/utils/drag-and-drop/draggable/draggable.ts

```diff
@@ -72,6 +72,15 @@ export class ClrDraggable<T> implements AfterContentInit, OnDestroy {
     this.dragEventListener.group = value;
   }
 
+  @Input('clrDragStartDelay')
+  set dragStartDelay(value: number) {
+    if (typeof value === 'number') {
```

Comment:
In that case, TS will complain that `value` is not a string. So you have to cast its type to string, which I am trying to avoid.

Reply:
Then a better solution it will add `string` as part of the possible types 

```ts
function dragStartDelay(value: number | string) {
  return parseInt(value as string, 10) || 0
}
dragStartDelay("100");
dragStartDelay(100);
```

When TypeScript compiles the code it will remove the type annotation and leave the code as simple as it could be.

```js
"use strict";
function dragStartDelay(value) {
    return parseInt(value, 10) || 0;
}
dragStartDelay("100");
dragStartDelay(100);
```

For example here is the instruction when we branch it. 
```bash
   13 E> 0x3e49bd357fca @    0 : a5                StackCheck
   25 S> 0x3e49bd357fcb @    1 : 25 02             Ldar a0
         0x3e49bd357fcd @    3 : 73 00             TestTypeOf #0
         0x3e49bd357fcf @    5 : 99 05             JumpIfFalse [5] (0x3e49bd357fd4 @ 10)
   62 S> 0x3e49bd357fd1 @    7 : 25 02             Ldar a0
   75 S> 0x3e49bd357fd3 @    9 : a9                Return
   85 S> 0x3e49bd357fd4 @   10 : 25 02             Ldar a0
         0x3e49bd357fd6 @   12 : 73 01             TestTypeOf #1
         0x3e49bd357fd8 @   14 : 99 14             JumpIfFalse [20] (0x3e49bd357fec @ 34)
  122 S> 0x3e49bd357fda @   16 : 13 00 00          LdaGlobal [0], [0]
         0x3e49bd357fdd @   19 : 26 fb             Star r0
         0x3e49bd357fdf @   21 : 0c 0a             LdaSmi [10]
         0x3e49bd357fe1 @   23 : 26 f9             Star r2
  129 E> 0x3e49bd357fe3 @   25 : 5e fb 02 f9 02    CallUndefinedReceiver2 r0, a0, r2, [2]
         0x3e49bd357fe8 @   30 : 96 03             JumpIfToBooleanTrue [3] (0x3e49bd357feb @ 33)
         0x3e49bd357fea @   32 : 0b                LdaZero
  153 S> 0x3e49bd357feb @   33 : a9                Return
         0x3e49bd357fec @   34 : 0d                LdaUndefined
  158 S> 0x3e49bd357fed @   35 : a9                Return
```

And here is the simple just `return parseInt(value, 10) || 0`

```bash
   13 E> 0x3e60f82d7fca @    0 : a5                StackCheck
   25 S> 0x3e60f82d7fcb @    1 : 13 00 00          LdaGlobal [0], [0]
         0x3e60f82d7fce @    4 : 26 fb             Star r0
         0x3e60f82d7fd0 @    6 : 0c 0a             LdaSmi [10]
         0x3e60f82d7fd2 @    8 : 26 f9             Star r2
   32 E> 0x3e60f82d7fd4 @   10 : 5e fb 02 f9 02    CallUndefinedReceiver2 r0, a0, r2, [2]
         0x3e60f82d7fd9 @   15 : 96 03             JumpIfToBooleanTrue [3] (0x3e60f82d7fdc @ 18)
         0x3e60f82d7fdb @   17 : 0b                LdaZero
   56 S> 0x3e60f82d7fdc @   18 : a9                Return
```

I think in the case above, the annotation is only helping you when you develop but won't prevent you from doing anything stupid.  But branching the code just to force the `tsc` to accept it is a bit overkill for this super small getter.

Reply:
That's what I am avoiding: casting string type to the value just to use it in `parseInt`. And to do that, you have to change the accepted types to both `string` and `number`. I am not supporting string value intentionally there to prevent confusion. It should be clear that It only accepts a numeric value. But it's reasonable for users to expect the following to work: `clrDragStartDelay="500"` This is handled properly in its current implementation without suggesting we also accept strings. That's more important than having one-liner there. And I am not worried about how TS compiles and does its job in this case.

Reply:
It's not one-liner the point here. If you require `number` - require `number`, adding annotation for number, but handle all other types inside the method is a bit of a lie and defeats the purpose of types. 

The idea of types is to give you information about what a function will accept/return and work with. I see and I get why `number` as argument of component is done this way. There are a lot of cases like that when TypeScript creates a fake feeling of safety. And keep in mind that TS is working only until we run `tsc` everything after that is just vanilla JS. So this method will and may get everything from `string`, `number`, `object`, `array` whatever and nothing could stop it ;)

But we don't need to handle all of them, we expect `number` and will parse `strings` to number just to not force people to wrap everything into `[input]`. So why we don't annotate it - what it is? Is more or less like an easter egg, "Oy, it could work with `string` as well, d*mn I did not know that"

```ts
parseInt(value as string, 10)
``` 

This type of casting is not really adding anything to the code after all is used only to tell `tsc` ok boy, don't throw an error. It's excluded from the compiled code.

```js
parseInt(value, 10)
```

But branching with `if/else` and adding `typeof` as operations are having an effect on the code. 
I'm ok with a lot of things but don't see why we need to write JavaScript code to make some TypeScripts rules pass that are adding overhead but don't give anything in return. (The reason for the if is that `parseInt` is annotated that the first argument must be `string` but .. it will work with `number` and just will return the number - again misleading annotation )

The speed improvement is not that big https://jsperf.com/if-else-parseint/1 but we did a lot more for a lot less. Still think that `parseInt` only is a better solution, providing readability and cover a lot more with less. 

I think that I prove my point so at least for me .. will stop bragging about it. ;)